AWSCore/Serialization/AWSURLRequestSerialization.m

```diff
@@ -93,12 +93,16 @@ - (AWSTask *)serializeRequest:(NSMutableURLRequest *)request
     NSDictionary *actionHTTPRule = [actionRules objectForKey:@"http"];
     NSString *ruleURIStr = [actionHTTPRule objectForKey:@"requestUri"];
 
+    NSDictionary *actionEndpoint = [actionRules objectForKey:@"endpoint"];
+    NSString *endpointHostPrefix = [actionEndpoint objectForKey:@"hostPrefix"];
```

Comment:
I hope this does not crash if `actionEndpoint` is `nil`

Reply:
I wouldn't expect a crash, but i would expect that sending `objectForKey:` to `actionEndpoint`  when `actionEndpoint` is `nil` will result in `endpointHostPrefix` being nil